### ISSUE TITLE
<vaadin-form-item> is breaking accessibility by not handling labels properly

### ISSUE THREAD

**Issue by limonte** [vaadin/vaadin-form-layout#74]
Live demo: https://jsfiddle.net/6r5htd05/280/

`<vaadin-date-picker>` inside `<vaadin-form-item>` doesn't have label pronounced

Confirmed with JAWS and NVDA.

**Comment by limonte** [comment]
This looks critical to me, `<vaadin-form-item>` is the perfect way to completely destroy forms a11y. Should we prioritize fixing this? @tomivirkki

**Comment by platosha** [comment]
This is really hard to implement properly until either of [`<vaadin-text-field-container>`](https://github.com/vaadin/vaadin-text-field/issues/67) or [AOM](https://github.com/WICG/aom) is shipped.

We should rather document that the form item label is visual only, so that users don’t expect the accessible label assignment from it.

Since the form item does not provide the accessible label assignment, I would suggest to use a regular `label` property of the components to set the accessible label on them. We were discussing adding means to visually hide the regular `label` inside the components, so that the is no duplicate label in cases like this:

```html
<vaadin-form-item>
  <label slot="label">Pick a date</label>
  <vaadin-date-picker label="Pick a date" hide-label></vaadin-date-picker>
</vaadin-form-item>
```

**Comment by samiheikki** [comment]
This is party fixed with #95 as we are promoting to use `vaadin-text-field` and `vaadin-date-picker` instead of `vaadin-form-item`

**Comment by rolfsmeds** [comment]
I see three possible solutions to this:

1. Provide accessible labels to the fields with `aria-label` (see https://github.com/vaadin/vaadin-text-field/issues/454)
2. Refactor all input field components to work with slotted `<input>` elements, and recommend that when using `vaadin-form-item`.
3. Deprecate `vaadin-form-item` and provide theme variants for input fields that move labels to the left.
